.github/workflows/roman_ci.yml

```diff
@@ -1,68 +1,109 @@
 # This workflow will install Python dependencies, run tests and lint with a variety of Python versions
 # For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
 
-name: Roman CI
+name: CI
 
 on:
   push:
-    branches: [ main ]
+    branches:
+      - main
+      - '*.x'
+    tags:
+      - '*'
   pull_request:
-    branches: [ main ]
+    branches:
+  schedule:
+    # Weekly Monday 9AM build
+    # * is a special character in YAML so you have to quote this string
+    - cron: '0 9 * * 1'
 
 env:
-  CRDS_SERVER_URL: "https://jwst-crds.stsci.edu"
+  CRDS_SERVER_URL: "https://roman-crds-dev.stsci.edu"
   CRDS_PATH: "/tmp/crds_cache"
+  CRDS_CLIENT_RETRY_COUNT: 3
+  CRDS_CLIENT_RETRY_DELAY_SECONDS: 20
 
 jobs:
-  build:
-
-    runs-on: ubuntu-latest
+  tox:
+    name: ${{ matrix.name }}
+    runs-on: ${{ matrix.os }}
     strategy:
+      fail-fast: false
       matrix:
-        python-version: ['3.8']
+        include:
+          - name: Code style checks
+            os: ubuntu-latest
+            python: 3.x
+            toxenv: codestyle
+
+          - name: Python 3.8
+            os: ubuntu-latest
+            python: 3.8
+            toxenv: py38-test
+
+          - name: Python 3.9
+            os: ubuntu-latest
+            python: 3.9
+            toxenv: py39-test
+
+          - name: Python 3.8 with dev dependencies
+            os: ubuntu-latest
+            python: 3.8
+            toxenv: py38-test-devdeps
+
+          - name: Python 3.8 with Numpy 1.20
+            os: ubuntu-latest
+            python: 3.8
+            toxenv: py38-test-numpy120
+
+          - name: macOS
+            os: macos-latest
+            python: 3.9
+            toxenv: py39-test
+
+          - name: Security audit
+            os: ubuntu-latest
+            python: 3.9
+            toxenv: bandit
+
+          - name: Installed package with --pyargs
+            os: ubuntu-latest
+            python: 3.8
+            toxenv: pyargs
+
+          - name: Build documentation and check warnings
+            os: ubuntu-latest
+            python: 3.8
+            toxenv: build_docs
+
+          - name: Latest dependency versions w/coverage
+            os: ubuntu-latest
+            python: 3.8
+            toxenv: py38-cov
 
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Python ${{ matrix.python-version }}
+    - name: Checkout code
```

Comment:
Over in `jwst`, before checking out the code, we do some package installation for the docs:

https://github.com/spacetelescope/jwst/blob/fec12a0e8eda7f930c33b8f0efaa440b5533a53f/.github/workflows/ci.yml#L105-L108

The conditional makes sure this only happens for tox env names that have "docs" in them.  This was cribbed from the way astropy did it until they switched to having a RTD integration as a check instead of doing it from the GA workflow.